### ISSUE TITLE
[BUG] Problems with enemy probabilities

### ISSUE THREAD

**Issue by goofygoober719** [TheAnsuz/Lethal-Company-Configurable-Company-API#44]
After playing around quite a bit with the mod, I recently noticed that certain enemy probability weights appear to be consistently breaking, and do not match the values set on the configuration menu.

To demonstrate, I set the probability value of every vanilla indoor enemy to 90. I also increased the max count of all enemies so nothing should interfere with the probabilities during the test.
The left screenshot is the first spawn cycle, and the right screenshot is a spawn cycle from around 10pm, near the end of the day.
![Probabilities](https://github.com/user-attachments/assets/8cf37fee-a66c-4c6d-9365-e59a940b4b91)
While most enemies follow the configured values correctly, these are the issues I found:
Ghost Girl - Below the configured value, but increases to the proper value before the end of the day
Hygrodere - Below the configured value
Thumper - Below the configured value
Jester - Below the configured value
Masked - Below the configured value, significantly increases over the course of the day
Barber - Below the configured value, significantly decreases over the course of the day

I then ran a second test with each value set to 5 rather than 90, and the same pattern can be seen
![Probabilities2](https://github.com/user-attachments/assets/b0bf8ba6-aee6-49bc-8a28-e122eeaa8ef7)
Both of my examples are from Artifice however I also tested this on other moons and found the same patterns each time.

- Game version: v56
- Configurable company: v3.4.0
- LethalCompanyVariables: v2.4.0
Profile code: 0190e9d4-3a82-6ef7-b8d2-b91b64530fe3

I am unaware if a similar thing happens with any outdoor entities, as the console doesn't provide much information for those spawns compared to the indoor spawns. However, I have noticed that the kidnapper fox has been able to spawn despite setting its value to 0 on all moons. I also noticed that the masked enemy in particular is not listed by the console for any moons that could not spawn it by default, but this does not appear to be an issue for any other entities.

**Comment by TheAnsuz** [comment]
Hello @goofygoober719 

Enemy spawning is a weird and complex system, there are two instances of the kidnapper fox spawning methods, one (the one that LCV uses) just would make it appear as normally meanwhile the game does not use this value as it spawns from bushes.

For the values being slightly different, LCV lets you modify the raw value of the enemy and even add new enemies to levels, however vanilla rules still apply. These rules include, but are not limited to:
- Enabled spawning (overrided by LCV)
- Level enemy power limitations (can be overrided by LCV)
- Enemy capacity reached (can be overrided by LCV)
- Enemy spawning falloff
- Level probability curve
- Enemy probability curve
- Time of day
- Number spawned

So it can vary slightly because of those values, the ones that LCV does not handle are because the game can break a lot of stuff if they are wrongly manipulated, yet most of the time you would only require to modify capacity and rarity.

To summarize, slighly deviations are possible but the overall spawning should not be pretty much affected as most of it are for vanilla enemy balancing. For the kidnapper, it uses a different approach for spawning from bushes that Im working to add in the second part of the LCV update.

Hope this information helps!

**Comment by goofygoober719** [comment]
Thank you for the info, I actually had no idea that these offsets occurred in the base game, but I guess now I know why I never seem to see the Barber. Is there any way an optional feature could be added to make the enemy probabilities completely static? (Aside from enemy probabilities being set to 0 when the max count or max power has been reached) I would be really interested in something like that, but I can understand if it isn't something that can be done, especially if tampering with it would break the game. Either way, thanks a lot for your help!

**Comment by TheAnsuz** [comment]
Hello @goofygoober719 

I am not aware of all the requirements for the new enemies barber and kidnapper fox. On behalf of the rest of enemies, those offsets are not that relevant to the gameplay, I'll still plan to add the suggested feature but for now keep in mind that you should really only care about the "distance" between probabilities of enemies, it is the same if all enemies are set to 90 than 10.

Will plan the feature, thanks for the report!
